### PR TITLE
Update stream BBEs

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
The original "streams" sample includes a streaming related scenario.

This PR renames the original "streams" sample as "streams-filtering" and introduces a new "streams" sample which demonstrates the stream type - possible ways in which streams could be declared and used.

### CODE REVIEW

File: examples/streams-filtering/streams-filtering.bal [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    // Write a query to filter out the teachers who are older than 18 years, wait until three teacher
+    // object are collected by the stream, group the 3 teachers based on their marital status, and calculate the
+    // unique marital status count of the teachers.
+    // Once the query is executed, publish the result to the `filteredStatusCountStream` stream.
+    forever{
+        from teacherStream where age > 18 window lengthBatch(3)
+        select status, count(status) as totalCount
+        group by status
+        having totalCount > 1
+        => (StatusCount [] status) {
```
Comment (afkham) [inline examples/streams-filtering/streams-filtering.bal]: StatusCount[] (should remove the space)

File: examples/streams/streams.bal [hunk 2]
```diff
@@ -1,70 +1,74 @@
...
 }
 
+//Function used to subscribe to a stream accepting float events.
+function printTemperature(float temperature) {
+    io:println("Temperature event received: " + temperature);
+}
+
+//Function used to subscribe to a stream accepting events of any type.
+function printEvent(any event) {
+    io:println("Event received: " + <string>event);
```
Comment (afkham) [inline examples/streams/streams.bal]: Now, io:println("Event received: ", <string>event); 
should work

**Review by afkham — CHANGES_REQUESTED** [review]
Minor changes

File: examples/streams-filtering/streams-filtering.bal (hunk 1 above)
Comment (MaryamZi) [reply examples/streams-filtering/streams-filtering.bal]: Fixed in 70258d2

File: examples/streams/streams.bal (hunk 2 above)
Comment (MaryamZi) [reply examples/streams/streams.bal]: Fixed in 70258d2

**Review by afkham — APPROVED** [review]

**Pull request merged**